src/com/sap/piper/PiperGoUtils.groovy

```diff
@@ -0,0 +1,60 @@
+package com.sap.piper
+
+class PiperGoUtils implements Serializable {
+
+    private static Script steps
+    private static Utils utils
+
+    private static String DELIMITER = '-DeLiMiTeR-'
+
+    PiperGoUtils(Script steps) {
+        this.steps = steps
+        this.utils = new Utils()
+    }
+
+    PiperGoUtils(Script steps, Utils utils) {
+        this.steps = steps
+        this.utils = utils
+    }
+
+    void unstashPiperBin() {
+
+        if (utils.unstash('piper-bin').size() > 0) return
+
+        def libraries = getLibrariesInfo()
+        String version
+        libraries.each {lib ->
+            if (lib.name == 'piper-lib-os') {
+                version = lib.version
+            }
+        }
+
+        def fallbackUrl = 'https://github.com/SAP/jenkins-library/releases/latest/download/piper_master'
```

Comment:
The approach here is to have an additional artifact always in the latest release which in fact does not really belong to that release but always to the current master.

I'm not fully against this appraoch. But wouldn't it be less misleading to have some kind of a "meta" release which always provides the binary matching to `master` rather than using the `latest` release for that. I guess it will cause confusion having an artifact in a release which does not really belong to that release.

Reply:
OK with me as well, we could also go for a dedicated release, e.g.  "MASTER", 

We productively use the other way already though in another place.
Thus in case we decide to do it differently now we should change it everywhere ...

@nevskrem @CCFenner @daniel-kurzynski what is your opinion?

Reply:
Any other opinion? If not I suggest to go ahead as is ...

Reply:
@OliverNocon I have no strong opinion on that one to be honest. My first reaction to the proposal was also like "well sounds a bit uncommon" but is also offers some "simplicity" by allowing to simply fetch the latest master artifact from the latest release.

Reply:
> but this approach also offers some "simplicity"

Having a dedicated meta release for the master version of the binary is IMO less confusing - and with that more simple - that having the current master version derived from the latest release. With that there is a clear separation between the releases and the current master version.

As mentioned already above I'm not fully against this approach, but I personally think having a dedicated master release is better (since less confusing).

Reply:
I kept it for now as is. We are still able to change it once we gathered some experience.

After having another thought about it I feel it is simpler for end-user to find the latest binary when we go via the latest release.
Looking at an old release with a different name which does not show up in the overview at first sight is not too intuitive. I think that it will likely lead to the situation that people will not use the master version at all - simply because they don't find it in the releases ...

I would opt for collecting some experiences first with the approach as implemented and then maybe re-adjust.